projects/shared/src/lib/view/thumbnail/thumbnail.component.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Component, Input, OnInit } from '@angular/core';
+import { of } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+
+@Component({
+  selector: 'shared-thumbnail',
+  templateUrl: './thumbnail.component.html',
+  styleUrls: ['./thumbnail.component.scss']
+})
+export class ThumbnailComponent implements OnInit {
+
+  /** Cover url */
+  coverUrl: string;
+
+  /** ISBN of the record */
+  isbn: string;
+
+  /** Css classes for the image */
+  imageCssClass = 'img-responsive img-thumbnail border border-light';
+
+  /** Style for image container */
+  figureStyle = 'thumb-detail';
+
+  /** Record to display */
+  @Input() record: any;
+
+  /** Should the thumbnail be small ? */
+  @Input() isSmall = false;
+
+  /**
+   * Constructor
+   * @param _httpClient - HttpClient
+   */
+  constructor(
+    private _httpClient: HttpClient
+  ) { }
+
+  /**
+   * Get cover image
+   * @returns cover image
+   */
+  getCoverImage() {
+    const url = `/api/cover/${this.isbn}`;
+    return this._httpClient.get<any>(url).pipe(
+      catchError(e => {
+        if (e.status === 404) {
+          return of(null);
+        }
+      })
+    );
+  }
+
+  /**
+   * Get cover url
+   * @returns string - url of the cover if cover exists.
+   */
+  getCoverUrl() {
```

Comment:
Group the 2 functions into 1: `getCoverImage` and `getCoverUrl` is overkill

Reply:
Done